matplotcheck/raster.py

```diff
@@ -100,68 +97,62 @@ def assert_legend_accuracy_classified_image(
         Returns
         ----------
         Nothing (if checks pass) or raises error
-
-
-        Notes
-        ----------
-        First compares all_label_options against the legend labels to find
-        which element of all_label_options matches that entry. E.g. if the
-        first legend entry has a match in the first list in all_label_options,
-        then that legend entry corresponds to the first class (value 0).
-        Then the plot image array is copied and the values are set to the
-        legend label that match the values (i.e. the element in
-        all_label_options). The same is done for the expected image array.
-        Finally those two arrays of strings are compared. Passes if they match.
         """
         # Retrieve image array
-        im_data = []
-        if self.ax.get_images():
-            im = self.ax.get_images()[0]
-            im_data, im_cmap = im.get_array(), im.get_cmap()
-        assert list(im_data), "No Image Displayed"
+        im_data = self.get_plot_image()
 
-        # Retrieve legend
-        legends = self.get_legends()
-        assert legends, "No legend displayed"
+        assert list(im_data), "No Image Displayed"
 
         # Retrieve legend entries and find which element of all_label_options
         # matches that entry
-        legend_dict = {}
```

Comment:
ok we need this functionality but it could be a little helper method which would be nice that returns a dictionary. i'd rather the dictionary be setup with the plot labels as the KEY, then the values would be 1. color and 2. true  or false so false would be if it doesn't match an expected label value and true means it matches an expected label value.